lib/ama_layout/navigation.rb

```diff
@@ -0,0 +1,33 @@
+module AmaLayout
+  class Navigation
+    include ActiveModel::Model
+    include Draper::Decoratable
+
+    attr_accessor :user, :current_url
+
+    cattr_accessor :member do
+      "member"
+    end
+    cattr_accessor :non_member do
+      "non-member"
+    end
+    cattr_accessor :member_in_renewal do
+      "member-in-renewal"
+    end
+    cattr_accessor :member_in_renewal_late do
+      "member-in-renewal-late"
+    end
+
+    def items
+      navigation_items.fetch(user.try(:navigation), []).map do |n|
+        NavigationItem.new n.merge({ current_url: current_url})
```

Comment:
Each nav item needs to know the current irk so we can appropriately expand
and/or highlight the nav items

On Mon, Aug 24, 2015 at 7:55 AM Ruben Estevez <notifications@github.com>
wrote:

> In lib/ama_layout/navigation.rb
> <https://github.com/amaabca/ama_layout/pull/18#discussion_r37753021>:
>
> > +    cattr_accessor :member do
> > +      "member"
> > +    end
> > +    cattr_accessor :non_member do
> > +      "non-member"
> > +    end
> > +    cattr_accessor :member_in_renewal do
> > +      "member-in-renewal"
> > +    end
> > +    cattr_accessor :member_in_renewal_late do
> > +      "member-in-renewal-late"
> > +    end
> > +
> > +    def items
> > +      navigation_items.fetch(user.try(:navigation), []).map do |n|
> > +        NavigationItem.new n.merge({ current_url: current_url})
>
> why do you have to merge the current url?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amaabca/ama_layout/pull/18/files#r37753021>.
>